Add expeditor

tduffield opened the pull request:
### Description

Expeditor is the next-generation utility that we'll use for version bumping. It will include additional features that we don't currently have like the ability to not bump a version on a specific PR.

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

lamont-granquist commented on .expeditor/bundle_install.sh:
```diff
@@ -0,0 +1,10 @@
+gem install bundler -v $bundler_version --user-install --conservative
+# WITH: ci (for version bumping and changelog creation)
+export BUNDLE_WITHOUT=omnibus_package:test:pry:integration:docgen:maintenance:travis:aix:bsd:linux:mac_os_x:solaris:windows:development
+bundle _${bundler_version}_ install
```
just do `bundle install` here -- you've already asserted the bundler version you're going to be running is the correct one, and this seems to get into a fight with the 1.14.x trampolining code.

lamont-granquist commented on .expeditor/bundle_install.sh:
```diff
@@ -0,0 +1,10 @@
+set -evx
+
+gem environment
+bundler_version=$(grep bundler omnibus_overrides.rb | cut -d'"' -f2)
```
this should probably be renamed because bundler uses `ENV['BUNDLER_VERSON']` now to setup trampolining, but that code is buggy and we don't want to use it, at at a minimum this is going to cause confusion to humans later.

tduffield replied:
so use a variable name other than bundler_version?

lamont-granquist replied:
the appveyor config used `$env:BUNDLER_VERSION=` which actively failed on bundler 1.14.x and i changed that to `$env:OMNIBUS_BUNDER=` here:

https://github.com/chef/chef/commit/a3159a69242adc94050e1664a7c1ced9d08a84a6#diff-180360612c6b8c4ed830919bbb4dd459

lamont-granquist replied:
yeah just rename the variable

lamont-granquist commented on tasks/version.rb:
```diff
@@ -39,3 +39,27 @@
   end
 end
+
+task :expeditor_update_version do
```
i don't know if this is a stub, but it doesn't seem to be useful, and it seems like the caller should be responsible for swallowing/ignoring exceptions directly.

tduffield replied:
When you say "this", what do you mean?

lamont-granquist replied:
the whole block of green code here.  if expeditor wants to call the rake tasks and ignore errors it should just call the rake tasks and ignore errors, rather than building additional rake tasks that ignore errors...

tduffield commented:
@coderanger I've applied many of the same principles we spoke about in chef/chef-dk#1213 to this PR as well. 

@lamont-granquist - Let me know if this new PR addresses your feedback :)
